standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5156,14 +5155,40 @@ public boolean drop_partition(final String db_name, final String tbl_name,
         null);
   }
 
-  private static class PathAndPartValSize {
-    PathAndPartValSize(Path path, int partValSize) {
-      this.path = path;
-      this.partValSize = partValSize;
+    /** Stores a path and its size. */
+    private static class PathAndDepth implements Comparable<PathAndDepth> {
+
+      final Path path;
+      final int depth;
+
+      public PathAndDepth(Path path, int depth) {
+        this.path = path;
+        this.depth = depth;
+      }
+
+      @Override
+      public int hashCode() {
+        return Objects.hash(path.hashCode(), depth);
+      }
+
+      @Override
+      public boolean equals(Object o) {
+        if (o == this) {
+          return true;
+        }
+        if (!(o instanceof PathAndDepth)) {
+          return false;
+        }
+        PathAndDepth p = (PathAndDepth) o;
+        return path.equals(p.path) && depth == p.depth;
```

Comment:
we can just use auto-generated `equals` method and should handle null here too:
```java
      @Override
      public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PathAndDepth that = (PathAndDepth) o;
        return depth == that.depth && Objects.equals(path, that.path);
      }
```

Reply:
Done.